File: backend-springboot/src/main/java/com/greenalgeria/damagereport/application/command/CreateDamageReportCommand.java
package com.greenalgeria.damagereport.application.command;

import com.greenalgeria.damagereport.application.*;
import com.greenalgeria.shared.cqrs.Command;

public record CreateDamageReportCommand(CreateDamageReportRequest request, String reportedBy)
        implements Command<DamageReportResponse> {

    public CreateDamageReportCommand(CreateDamageReportRequest request) {
        this(request, request.reportedBy());
    }
}

File: backend-nestjs/src/modules/damage-reports/application/commands/create-damage-report.command.ts
import { Command } from '@nestjs/cqrs';
import { DamageReport } from '../../domain/damage-report';
import {
  type DamageReportType,
  type DamageReportSeverity,
} from '../../domain/damage-report.types';

export class CreateDamageReportCommand extends Command<DamageReport> {
  constructor(
    readonly zoneId: string,
    readonly type: DamageReportType,
    readonly severity: DamageReportSeverity,
    readonly lat: number,
    readonly lng: number,
    readonly description: string,
    readonly reportedBy: string,
  ) {
    super();
  }
}
